release notes/v0.3.0.md

```diff
@@ -0,0 +1,64 @@
+xk6-browser v0.3.0 is here! :tada:
+
+This is a minor release with some important bug fixes, a couple of new features, improvements and code refactors!
+
+
+## New features
+
+- Implemented `Browser.on(event)`. ([#268](https://github.com/grafana/xk6-browser/pull/268), [#287](https://github.com/grafana/xk6-browser/pull/287))
+
+  `on('disconnected')` returns a `Promise` you can wait on to get notified when the WebSocket connection to the browser is closed. See [this example](https://github.com/grafana/xk6-browser/blob/v0.3.0/examples/browser_on.js) for details.
+
+  This is the first async method in xk6-browser, using the newly released [event loop support in k6](https://github.com/grafana/k6/pull/2228), and marks the start of our transition to async APIs. :tada:
+
+- Exposed preconfigured mobile device emulation settings. ([#289](https://github.com/grafana/xk6-browser/pull/289))
+
+  We maintain an internal list of popular devices with their viewport size, user agent and scaling factor, which is useful to test how a web site looks and behaves on different devices. See [this example](https://github.com/grafana/xk6-browser/blob/v0.3.0/examples/device_emulation.js) for details.
+
+
+
+## Bugs fixed
+
+- Made concealed elements clickable. ([#264](https://github.com/grafana/xk6-browser/pull/264))
+
+  Previously, interacting with an element obscured by another was failing with `wait for selector did not result in any nodes`. Now we properly detect this scenario and retry the action with different scroll positions. This affects all `Element` and `Frame` `click()`, `dblclick()`, `hover()`, `check()`, `uncheck()` and `tap()` actions.
+
+- Fixed `waitForSelector()` flakiness by retrying once on failure. ([#260](https://github.com/grafana/xk6-browser/pull/260))
+
+- Fixed data race accessing `Frame.documentHandle` while switching execution contexts. ([#263](https://github.com/grafana/xk6-browser/pull/263))
+
+- Fixed keyboard press of special keys. ([#283](https://github.com/grafana/xk6-browser/pull/283))
+
+  Previously pressing some keys like Backspace, Delete, ArrowLeft and AltGraph would panic because of a wrong CDP argument.
+
+- Fixed negative metric emission in certain rare scenarios. ([#288](https://github.com/grafana/xk6-browser/pull/288))
+
+- Fixed `Page.waitForFunction()`. ([#294](https://github.com/grafana/xk6-browser/pull/294))
+
+  This was previously broken and untested, and with the fix all polling variants (`requestAnimationFrame`, DOM mutation and interval) should work correctly. See [this example](https://github.com/grafana/xk6-browser/blob/v0.3.0/examples/waitforfunction.js) for details.
```

Comment:
Instead of using negative language (which may give a wrong impression), would it be better if we just say the following?

```suggestion
  `waitForFunction` wasn't reliable. This fix reimplements it from scratch and adds proper support for the polling options. The polling options can be one of `raf`, `mutation`, or a number (defaults to `raf`). See [this example](https://github.com/grafana/xk6-browser/blob/v0.3.0/examples/waitforfunction.js) for details.
```

Reply:
It's not that `waitForFunction` wasn't reliable—it's that it didn't work at all (it would panic). It's also not a "from scratch" reimplementation, but more of a rewrite with a bunch of fixes.

But I'll change this to soften the tone and mention the proper valid options. :+1:

Reply:
Let me know if ad0d4d9 resolves this.

Reply:
Nice write-up. Would it be better if we also add that polling defaults to `raf`?

Reply:
Sure, done in 02ebb34.